File: arrays/hm.py
def twoSum(nums, target):
    hm = {}
    for index, value in enumerate(nums):
        diff = target - nums[index]
        if diff in hm:
            return [hm[diff], index]
        hm[value] = index

print(twoSum([2, 7, 9], 9))

File: hm.rb
def hm(nums, target)
  hm = {}
  nums.each |num| do
    diff = target - num
    if diff in hm
      return hm[diff], num
    hm[num] = diff
    end
  end
end

puts hm([1,2,4,5], 9)
